homeassistant/components/econet/sensor.py

```diff
@@ -1,113 +1,143 @@
-    POWER_USAGE_TODAY: "todays_energy_usage",
-    ALERT_COUNT: "alert_count",
-    WIFI_SIGNAL: "wifi_signal",
-    RUNNING_STATE: "running_state",
-}
-
-SENSOR_NAMES_TO_UNIT_OF_MEASUREMENT = {
-    TANK_HEALTH: PERCENTAGE,
-    AVAILABLE_HOT_WATER: PERCENTAGE,
-    COMPRESSOR_HEALTH: PERCENTAGE,
-    OVERRIDE_STATUS: None,
-    WATER_USAGE_TODAY: UnitOfVolume.GALLONS,
-    POWER_USAGE_TODAY: None,  # Depends on unit type
-    ALERT_COUNT: None,
-    WIFI_SIGNAL: None,
-    RUNNING_STATE: None,  # This is just a string
-}
+
+ENERGY_KILO_BRITISH_THERMAL_UNIT = "kBtu"
 
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up EcoNet sensor based on a config entry."""
 
-    equipment = hass.data[DOMAIN][EQUIPMENT][entry.entry_id]
-    sensors = []
-    all_equipment = equipment[EquipmentType.WATER_HEATER].copy()
-    all_equipment.extend(equipment[EquipmentType.THERMOSTAT].copy())
+    data = hass.data[DOMAIN][EQUIPMENT][entry.entry_id]
+    equipment = data[EquipmentType.WATER_HEATER].copy()
+    equipment.extend(data[EquipmentType.THERMOSTAT].copy())
 
-    for _equip in all_equipment:
-        for name, attribute in SENSOR_NAMES_TO_ATTRIBUTES.items():
-            if getattr(_equip, attribute, None) is not None:
-                sensors.append(EcoNetSensor(_equip, name))
-        # This is None to start with and all device have it
-        sensors.append(EcoNetSensor(_equip, WIFI_SIGNAL))
+    sensors = []
 
-    for water_heater in equipment[EquipmentType.WATER_HEATER]:
-        # These aren't part of the device and start off as None in pyeconet so always add them
-        sensors.append(EcoNetSensor(water_heater, WATER_USAGE_TODAY))
-        sensors.append(EcoNetSensor(water_heater, POWER_USAGE_TODAY))
+    sensors = [
+        EcoNetSensor(_equip, description)
+        for _equip in equipment
+        for description in SENSOR_TYPES
+        if getattr(_equip, description.key, False) is not False
+    ]
 
     async_add_entities(sensors)
 
 
 class EcoNetSensor(EcoNetEntity, SensorEntity):
     """Define a Econet sensor."""
 
-    def __init__(self, econet_device, device_name):
+    entity_description: SensorEntityDescription
+
+    def __init__(
+        self,
+        econet_device,
+        description: SensorEntityDescription,
+    ) -> None:
         """Initialize."""
         super().__init__(econet_device)
+        self.entity_description = description
         self._econet = econet_device
-        self._device_name = device_name
 
     @property
     def native_value(self):
         """Return sensors state."""
-        value = getattr(self._econet, SENSOR_NAMES_TO_ATTRIBUTES[self._device_name])
+        value = getattr(self._econet, self.entity_description.key)
+        if self.entity_description.name == "power_usage_today":
+            if self._econet.energy_type == ENERGY_KILO_BRITISH_THERMAL_UNIT.upper():
```

Comment:
I think the constant can be dropped now:
```suggestion
            if self._econet.energy_type == "KBTU":
```